gammapy/analysis/core.py

```diff
@@ -292,11 +310,22 @@ def _spectrum_extraction(self):
 
         maker_config["selection"] = ["counts", "aeff", "edisp"]
         dataset_maker = SpectrumDatasetMaker(**maker_config)
-        bkg_maker_config = {}
-        if datasets_settings.background.exclusion:
-            exclusion_region = Map.read(datasets_settings.background.exclusion)
-            bkg_maker_config["exclusion_mask"] = exclusion_region
-        bkg_maker = ReflectedRegionsBackgroundMaker(**bkg_maker_config)
+
+        bkg_method = self.config.datasets.background.method
+        if bkg_method == "reflected":
+            bkg_maker_config = {}
+            if datasets_settings.background.exclusion:
+                exclusion_region = Map.read(datasets_settings.background.exclusion)
+                bkg_maker_config["exclusion_mask"] = exclusion_region
```

Comment:
Could we add the following line here, as it is done in the `map_making()' function ?
```
bkg_maker_config.update(self.config.datasets.background.parameters)
```